resources/prosody-plugins/mod_av_moderation_component.lua

```diff
@@ -0,0 +1,242 @@
+local is_healthcheck_room = module:require 'util'.is_healthcheck_room;
+local json = require 'util.json';
+local st = require 'util.stanza';
+local um_is_admin = require "core.usermanager".is_admin;
+
+local muc_component_host = module:get_option_string('muc_component');
+if muc_component_host == nil then
+    log('error', 'No muc_component specified. No muc to operate on!');
+    return;
+end
+
+module:log('info', 'Starting av_moderation for %s', muc_component_host);
+
+local function is_admin(jid)
+    return um_is_admin(jid, module.host);
+end
+
+-- Notifies that av moderation has been enabled or disabled
+-- @param jid the jid to notify, if missing will notify all occupants
+-- @param enable whether it is enabled or disabled
+-- @param room the room
+-- @param actorJid the jid that is performing the enable/disable operation
+-- @param mediaType the media type for the moderation
+function notify_occupants_enable(jid, enable, room, actorJid, mediaType)
+    local body_json = {};
+    body_json.type = 'av_moderation';
+    body_json.enabled = enable;
+    body_json.room = room.jid;
+
+    local notify = function(jid_to_notify, mediaType)
+        body_json.actor = actorJid;
+        body_json.mediaType = mediaType;
+        local body_json_str = json.encode(body_json);
+        local stanza = st.message({ from = module.host; to = jid_to_notify; })
+                :tag('json-message', { xmlns = 'http://jitsi.org/jitmeet' }):text(body_json_str):up();
+        module:send(stanza);
+    end
+
+    if jid then
+        notify(jid, mediaType);
+    else
+        for _, occupant in room:each_occupant() do
+            notify(occupant.jid, mediaType);
+        end
+    end
+end
+
+-- Notifies about a jid added to the whitelist. Notifies all moderators and admin and the jid itself
+-- @param jid the jid to notify about the change
+-- @param moderators whether to notify all moderators in the room
+-- @param room the room where to send it
+-- @param mediaType used only when a participant is approved (not sent to moderators)
+function notify_whitelist_change(jid, moderators, room, mediaType)
+    local body_json = {};
+    body_json.type = 'av_moderation';
+    body_json.room = room.jid;
+    body_json.whitelists = room.av_moderation;
+    local moderators_body_json_str = json.encode(body_json);
+    body_json.whitelists = nil;
+    body_json.approved = true; -- we want to send to participants only that they were approved to unmute
+    body_json.mediaType = mediaType;
+    local participant_body_json_str = json.encode(body_json);
+
+    for _, occupant in room:each_occupant() do
+        local body_to_send = nil;
+        if moderators and occupant.role == 'moderator' then
+            body_to_send = moderators_body_json_str;
+        elseif occupant.jid == jid then
+            body_to_send = participant_body_json_str;
+        end
+
+        if body_to_send then
```

Comment:
You use two different patterns to accomplish a very similar thing in `notify_occupants_enable` and `notify_whitelist_change`. It could be much simpler if you extract a utility to send a `body` encoded as a json message to a specific jid.

Reply:
Fixed.

Reply:
Now that you got this far it would be a shame not to further simplify :)
```
if moderators and occupant.role == 'moderator' then
  send_json_message(occupant.jid, moderators_body_json_str)
elseif occupant.jid == jid then
  send_json_message(occupant.jid, participant_body_json_str)
end
```